### PR TITLE
Enterprise Release - June 2020

### LINKED ISSUE #440: [Enterprise] Update Enterprise document pages for the new architecture
We're redesigning our internal architecture, so this documentation should be updated to follow it.

### PR DESCRIPTION
**Pull request by yykamei**
This release contains a critical change, so I want to rewrite the Enterprise document pages for this release.

I'm going to create pages with the title of the contents below.

- [x] System overview
    - [x] Application architecture
    - [x] Requirements
        - [x] Software Requirements
        - [x] Hardware Requirements
- [x] Installation
- [x] Configuring Sider Enterprise
    - [x] GitHub Enterprise Server Integration
    - [x] HTTP Proxy
    - [x] Runners
    - [x] Live Update
    - [x] SMTP
- [x] Maintaining Sider Enterprise
    - [x] Make a user admin
    - [x] Restart Sider Enterprise
    - [x] Collect debug log
    - [x] Metrics you should watch
- [x] Updating Sider Enterprise
- [x] Load Balancer Guide
- [x] MySQL Guide
    - [x] Recommended parameters you should configure
- [x] Redis Guide
- [x] Minio Guide
    - [x] Lifecycle
- [x] GitHub Enterprise Server Guide
- [x] Clustering Guide
- [x] Examples

Close #440

### CODE REVIEW

File: website/sidebars.json [hunk 1]
```diff
@@ -148,9 +138,16 @@
...
-    "News": ["news/2020", "news/2019"]
+    "Enterprise Releases": [
+      "enterprise/releases/index",
+      "enterprise/releases/202005",
+      "enterprise/releases/202004",
+      "enterprise/releases/202002",
+      "enterprise/releases/202001",
+      "enterprise/releases/201911",
+      "enterprise/releases/2018010801-201909.0"
+    ]
```
Comment (yykamei) [inline website/sidebars.json:151]: I extracted release notes of Sider Enterprise here because I couldn't collapse the subcategories.

File: docs/enterprise/config.md [hunk 2]
```diff
@@ -1,604 +1,191 @@
...
+## GitHub Enterprise Server
 
-### `ACTION_MAILER_SMTP_ENABLE_STARTSSL_AUTO` (Optional)
+You have to configure these parameters to integrate with GitHub Enterprise Server. After setting up your GitHub App, prepare these parameters carefully.
 
-Example:
+- `GITHUB_ENDPOINT` - (Required) The URL which points to the GitHub Enterprise web page. e.g., `https://github.example.com`.
 
-    ACTION_MAILER_SMTP_ENABLE_STARTSSL_AUTO=no
+- `GITHUB_API_ENDPOINT` - (Required) The URL which points to the GitHub Enterprise API endpoint. Basically, the suffix of this URL must be `/api/v3`. See [Endpoint URLs](https://developer.github.com/enterprise/v3/enterprise-admin/) to learn more about the GitHub Enterprise endpoints. e.g., `https://github.example.com/api/v3`.
```
Comment (yykamei) [inline docs/enterprise/config.md:64]: > __[remark-lint]__ Link to https://developer.github.com/enterprise/v3/enterprise-admin/ is dead ([view](https://sider.review/gh/repos/142577083/pulls/446#issue-4023295))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `no-dead-urls` | `warn` |
>
> You can [close this issue](https://sider.review/gh/repos/142577083/pulls/446#issue-4023295) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

This URL will redirect users to https://developer.github.com/enterprise/2.21/v3/enterprise-admin/. I think it's better to use the link without a version as long as GitHub redirects permanently.

File: docs/enterprise/config.md [hunk 3]
```diff
@@ -1,604 +1,191 @@
...
-Example:
+  ```console
+  tr -dc '[:alnum:]' < /dev/urandom | head -c 32
+  ```
 
-    EXCEPTION_NOTIFIER_RECIPIENT_EMAILS=foo@example.com,bar@example.com
+## Runners and Minio
 
-### `SECRET_KEY_BASE`
+sideci, which is a core component of Sider Enterprise, will invoke Runners to perform analyses. The parameters in this section control how Sider should invoke Runners, and how the traces generated by Runners should be stored. See [Minio Guiode](./minio.md) for the set up of Minio.
```
Comment (yykamei) [inline docs/enterprise/config.md:88]: > __[LanguageTool]__ This sentence does not start with an uppercase letter ([view](https://sider.review/gh/repos/142577083/pulls/446#issue-4023342))
>
> <details>
>
> | Rule | Type | Category | Replacements |
> | ---- | ---- | -------- | ------------ |
> | `UPPERCASE_SENTENCE_START` | typographical | CASING | `Sideci` |
>
> You can [close this issue](https://sider.review/gh/repos/142577083/pulls/446#issue-4023342) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

`sideci` should be written with lowercase.

File: docs/enterprise/mysql.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+- [`tmp_table_size`](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_tmp_table_size) and [`max_heap_table_size`](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_heap_table_size) - We recommend configuring these parameters with `268435456` or higher together. Note they depend on the memory of your MySQL server. You can increase the value more if the memory space is large.
+
+- [`innodb_temp_data_file_path`](https://dev.mysql.com/doc/refman/5.7/en/innodb-parameters.html#sysvar_innodb_temp_data_file_path) - The recommended value is `ibtmp1:12M:autoextend:max:5G`. Fine-tune the trailing parameter `5G` to fit your environment. This size is the maximum size of the temporary tablespace data.
+
+  The queries called by Sider Enterprise consume an in-memory temporary table, and it will be converted to an on-disk temporary table after the size exceeds the smaller of `tmp_table_size` and `max_heap_table_size`. The temporary table file size will be extended with `innodb_temp_data_file_path`, but the default value is `ibtmp1:12M:autoextend`, which means the file will grow without limitation. Eventually, this would cause out of disk.
```
Comment (yykamei) [inline docs/enterprise/mysql.md:26]: > __[LanguageTool]__ Use "smallest" if you mean to express an extreme with this adjective. ([view](https://sider.review/gh/repos/142577083/pulls/446#issue-4029949))
>
> <details>
>
> | Rule | Type | Category | Replacements |
> | ---- | ---- | -------- | ------------ |
> | `THE_WORSE_OF` | grammar | GRAMMAR | `smallest` |
>
> You can [close this issue](https://sider.review/gh/repos/142577083/pulls/446#issue-4029949) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I'm following the sentence [here](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_tmp_table_size).

**Comment by yykamei** [comment]
@ybiquitous @sumyapp Sorry, this PR has a large diff, but I want you to check the page structure and tell me the missing contents I should write.

File: docs/enterprise/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Sider Enterprise allows you to run Sider on your infrastructure. This allows Sider Enterprise to satisfy higher security and performance requirements than the SaaS version of Sider.
+
+- You can keep your source code in your network. While SaaS version of Sider offers a high-security environment, keeping your source code in your network can enable even higher security.
+- You can upgrade the hardware for better analysis performance as you need it.
+
+On the other hand, using Sider Enterprise means you have to take care of running the application. You have to set up the hardware and software required to run it and upgrade the programs.
+
+If you are using GitHub Enterprise, Sider Enterprise is the only way to use Sider on your projects.
+
+## [System overview](./system-overview.md)
```
Comment (ybiquitous) [inline docs/enterprise/index.md:19]: [suggestion] How about making a consistent style in headings? (we may need to fix other places...)

```suggestion
## [System Overview](./system-overview.md)
```

File: docs/enterprise/index.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+System overview of Sider Enterprise. You can get to know the requirements to run Sider Enterprise on your environment.
+
+## [Installation Guide](./installation.md)
```
Comment (ybiquitous) [inline docs/enterprise/index.md:23]: [suggestion] I feel the repeated `Guide` words are a bit wordy. What about removing "Guide" from the headings and siderbar labels?

For example:

image

Comment (ybiquitous) [reply docs/enterprise/index.md:23]: Also, I feel just `Configuration` is confusing with the normal configuration page of Sider.
So, what about adding `Enterprise` to `<title>` and `<h1>` as follows?

image

File: docs/enterprise/system-overview.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+- MySQL
+- Redis
+- Minio
+- GitHub Enterprise Server
+
+This is an architecture diagram that could help you understand the Sider Enterprise system.
+
+![Application Architecture](https://app.lucidchart.com/publicSegments/view/9499e669-2f00-4a45-9b58-200b51dce2f1/image.png)
+
+Sider Enterprise itself consists of two components: sideci and Runners. These two components will run on Docker.
```
Comment (ybiquitous) [inline docs/enterprise/system-overview.md:26]: [IMO] the name `sideci` looks a typo to me... How about expressing it as a code or emphasized?

`sideci` or **sideci**

File: docs/enterprise/config.md [hunk 8]
```diff
@@ -1,604 +1,191 @@
...
+- `ACTION_MAILER_SMTP_ENABLE_STARTSSL_AUTO` - (Optional) The boolean to control if Sider Enterprise uses STARTTLS. The default value is `true`.
 
-| Introduced | Deprecated |
-| ---------- | ---------- |
-| 201908     | -          |
+- `EXCEPTION_NOTIFIER_RECIPIENT_EMAILS` - (Optional) The comma-separated list of recipients for error reporting emails. e.g., `foo@example.com,bar@example.com`.
 
-32 bytes of random string to use for ENCRYPTION_SERVICE_KEY as a salt.
-You cannot change this value once you set up.
+## Live Update
```
Comment (ybiquitous) [inline docs/enterprise/config.md:144]: [suggestion] I feel `Live Update` is an online-update of Sider Enterprise itself. How about renaming as follows?

```suggestion
## Auto-Refresh Pages
```

File: docs/enterprise/config.md (hunk 3 above)
Comment (ybiquitous) [reply docs/enterprise/config.md:88]: [suggestion] How about rewriting it as follows?

```diff
-sideci, which is a core component of Sider Enterprise, will invoke
+The core component `sideci` will invoke
```

**Comment by yykamei** [comment]
Thank you for spending your time to read the PR :pray: I will see your review carefully.

File: docs/enterprise/index.md (hunk 6 above)
Comment (yykamei) [reply docs/enterprise/index.md]: Thank you. I removed `Guide` from all pages via 6aa94f0, and I prefixed "Sider Enterprise" via e352dc7. "Sider" might be wordy, but I added it before "Enterprise" because I want to distinguish from GitHub Enterprise Server.

File: docs/enterprise/system-overview.md (hunk 7 above)
Comment (yykamei) [reply docs/enterprise/system-overview.md:26]: I agree with you. I will make the word bold. In addition, I want to make "sideci web" and "sideci worker" bold, too.

File: docs/enterprise/index.md (hunk 6 above)
Comment (ybiquitous) [reply docs/enterprise/index.md]: > "Sider" might be wordy, but I added it before "Enterprise" because I want to distinguish from GitHub Enterprise Server.

Honestly, I feel a bit strange that "Sider Enterprise Configuration" and "Installation" are nearby in the sidebar. 💦

image

How about this change? 👇 

image

File: docs/enterprise/config.md (hunk 8 above)
Comment (yykamei) [reply docs/enterprise/config.md]: 178f277

File: docs/enterprise/system-overview.md (hunk 7 above)
Comment (ybiquitous) [reply docs/enterprise/system-overview.md]: Looks good for using the **bold** style. 👍 

If we want to increase searchability, the one word "sideci-web" may be better than `sideci web` (if possible).

File: docs/enterprise/index.md (hunk 6 above)
Comment (yykamei) [reply docs/enterprise/index.md]: Good catch :clap: I didn't notice. I intended to separate the commit, but I accidentally pushed at a time via 178f277 :pray:

Comment (ybiquitous) [reply docs/enterprise/index.md]: I feel also wordy "Sider Enterprise Configuration" in the outline page:

image

https://deploy-preview-446--sider-docs.netlify.app/enterprise/

The following may be simpler.

---

## Configuration

Configuration guide for Sider Enterprise. You may have to ...

File: docs/enterprise/system-overview.md (hunk 7 above)
Comment (yykamei) [reply docs/enterprise/system-overview.md]: OK, It's also great :+1:

Comment (yykamei) [reply docs/enterprise/system-overview.md]: Thank you so much. I could change it via 9ae324e.

File: docs/enterprise/index.md (hunk 6 above)
Comment (yykamei) [reply docs/enterprise/index.md]: Thank you. I think 016c1fe might change it.

**Comment by ybiquitous** [comment]
I think the pages' outline and structure are good. 👍 

In addition to this big change, what about improving a URL, sidebar label, and a title of the release pages? The following image is the current sidebar, but I feel it's not user-friendly:

image

My suggestions:

- Change sidebar labels like `202006` → `June 2020`
- Change `<title>` and `<h1>` like `Release Notes - 202006` → `Enterprise Release Note on June 2020`
- Remove needless `<h2>` like `release-202006.0`

image

But, these suggestions may be good to address in another pull request. 👍

File: docs/enterprise/index.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+title: Sider Enterprise Outline
+sidebar_label: Outline
+hide_title: true
+---
+
+# Sider Enterprise
+
+Sider Enterprise allows you to run Sider on your infrastructure. This allows Sider Enterprise to satisfy higher security and performance requirements than the SaaS version of Sider.
+
+- You can keep your source code in your network. While SaaS version of Sider offers a high-security environment, keeping your source code in your network can enable even higher security.
```
Comment (ybiquitous) [inline docs/enterprise/index.md:12]:
```suggestion
- You can keep your source code in your network. While the SaaS version of Sider offers a high-security environment, keeping your source code in your network can enable even higher security.
```

File: docs/enterprise/index.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+- You can upgrade the hardware for better analysis performance as you need it.
+
+On the other hand, using Sider Enterprise means you have to take care of running the application. You have to set up the hardware and software required to run it and upgrade the programs.
```
Comment (ybiquitous) [inline docs/enterprise/index.md:15]: [IMO] I feel it more natural, but just my opinion (you can ignore it).

```suggestion
On the other hand, using Sider Enterprise means you have to manage the system by yourself. You have to set up the hardware and software required to run it and upgrade the programs.
```

File: docs/enterprise/config.md (hunk 3 above)
Comment (yykamei) [reply docs/enterprise/config.md]: I overlooked your suggestion. Thank you. I changed via 06808c4.

**Comment by yykamei** [comment]
> * Remove needless `<h2>` like `release-202006.0`

It's good to update releases, but I won't remove `<h2>` because we may release `release-202006.1` in the same month for bugfixes. Thank you for your suggestion, and I will change the label of the sidebar and title of releases.

**Comment by ybiquitous** [comment]
> I won't remove `<h2>` because we may release `release-202006.1` in the same month for bugfixes.

Oh, I missed it. Please keep `<h2>` as a version. 👍

File: docs/enterprise/index.md (hunk 10 above)
Comment (yykamei) [reply docs/enterprise/index.md:15]: I prefer your suggestion. Thanks.

File: docs/enterprise/index.md (hunk 5 above)
Comment (yykamei) [reply docs/enterprise/index.md:19]: You're absolutely right.

File: docs/enterprise/example-deployments/single-node-with-docker-compose.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
+---
```
Comment (ybiquitous) [inline docs/enterprise/example-deployments/single-node-with-docker-compose.md:1]: [IMO] What about using `examples` simply?

```diff
-docs/enterprise/examples/single-node-with-docker-compose.md
+docs/enterprise/example-deployments/single-node-with-docker-compose.md
```

File: docs/enterprise/index.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+If you are using GitHub Enterprise, Sider Enterprise is the only way to use Sider on your projects.
+
+## [System Overview](./system-overview.md)
+
+System overview of Sider Enterprise. You can get to know the requirements to run Sider Enterprise on your environment.
+
+## [Installation](./installation.md)
+
+Step-by-step Installation guide for Sider Enterprise.
```
Comment (ybiquitous) [inline docs/enterprise/index.md:25]: [typo] 😅 

```suggestion
Step-by-step installation guide for Sider Enterprise.
```

File: docs/enterprise/index.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+MySQL guide for Sider Enterprise. It describes the requirements and the recommended parameters.
+
+## [Redis](./redis.md)
+
+Redis guide for Sider Enterprise.
+
+## [Minio](./minio.md)
+
+Minio guide for Sider Enterprise.
```
Comment (ybiquitous) [inline docs/enterprise/index.md:53]: [IMO] It may be good to add a simple note for people unfamiliar with Minio.

```suggestion
Minio (compatible with Amazon S3) guide for Sider Enterprise.
```

File: docs/enterprise/example-deployments/single-node-with-docker-compose.md (hunk 11 above)
Comment (yykamei) [reply docs/enterprise/example-deployments/single-node-with-docker-compose.md]: Thank you. It looks better. I changed via 210b9bf.

**Comment by yykamei** [comment]
Thank you. I didn't notice that MiniO is correct.

**Comment by ybiquitous** [comment]
I'll add MinIO to https://github.com/sider/goodcheck-rules 👍

File: docs/enterprise/index.md (hunk 13 above)
Comment (yykamei) [reply docs/enterprise/index.md]: Thank you :+1: fb632ec

**Comment by ybiquitous** [comment]
`MinIO`? 1c2b55a

**Comment by yykamei** [comment]
Thank you. It's too difficult for me to correctly type it. ea2cd8d

**Comment by yykamei** [comment]
I confirmed the next release, `release-202006.0`, worked on my staging environment. I will merge and announce tomorrow.
